chapter_10_external_events.asciidoc

```diff
@@ -55,22 +55,224 @@ events|in                                              events|out
+others. We want to have strong connascence locally, like when two classes are
+closely related, but weak connascence at a distance.
+
+In our first example of a distributed ball of mud, we see Connascence of
+Execution - multiple components need to know the correct order of work for an
+operation to be successful. 
+
+When thinking about error conditions here, we're talking about Connascence of
+Timing - multiple things have to happen one after another for the operation to
+work.
+
+When we replace our RPC style system with events, we replace both of these types
+of connascence with a _weaker_ type: Connascence of Name, where multiple
+components only need to agree on the name of an event, and the names of fields
+it carries.
+
+We can never completely avoid coupling, except by having our software not talk
+to any other software. What we want is to avoid _inappropriate_ coupling. 
+Connascence provides a mental model for understanding the strength and type of
+coupling inherent in different architectural styles. Read all about it at
+www.connascence.io
+*******************************************************************************
+
+
+Really we want to make sure that when we place an order, the order is placed.
+Everything else can happen _later_ so long as it happens. If that sounds
+familiar, it should! This is the same requirement we have of our aggregates and
+commands.
+
+Like aggregates, microservices should be _consistency boundaries_. Between two
+services, we can accept eventual consistency, and that means we don't need to
+rely on synchronous calls. Each service accepts commands from the outside world
+and raises events to record the result. Other services can listen to those
+events to trigger the next steps in the workflow.
+
 
 === Using a Redis Pubsub Channel for Integration
 
 To avoid the "distributed BBOM" antipattern, instead of temporally coupled HTTP
-API calls, we want to use some sort of asynchronous messaging layer to
-integrate between systems.  We want our "batch quantity changed" messages to
-come in as external events from upstream systems, and we want our system to
-publish "allocated" events for downstream systems to listen to.
-
-When moving towards events as an integration solution, you need to choose
-some sort of technology for passing those events from one system to another.
-We need to be able to publish events to some central service, and we need some
-way for other systems to be able to "subscribe" to different types of messages,
-and pick them up asynchronously from some sort of queue.
-
-At MADE.com we use https://eventstore.org/[Eventstore];  Kafka or RabbitMQ
+API calls, we want to use asynchronous messaging to integrate our systems. We
+want our "batch quantity changed" messages to come in as external messages from
+upstream systems, and we want our system to publish "allocated" events for
+downstream systems to listen to.
+
+TODO: This doesn't sit naturally here.
```

Comment:
how about a new subheading called `=== The Alternative: Temporal Decoupling using Asynchronous Messaging` or s'thing.  and then move the `Using a Redis Pubsub` down to just before "We'll need some way of getting events" para.

Reply:
did this.